app/lib/server/methods/leaveRoom.js

```diff
@@ -16,7 +17,11 @@ Meteor.methods({
 		const room = Rooms.findOneById(rid);
 		const user = Meteor.user();
 
-		if (room.t === 'd' || (room.t === 'c' && !hasPermission(user._id, 'leave-c')) || (room.t === 'p' && !hasPermission(user._id, 'leave-p'))) {
+		if (!roomTypes.getConfig(room.t).allowMemberAction(room, RoomMemberActions.LEAVE)) {
```

Comment:
We could pass the userId as a third parameter and check for the permissions there as well when userId is provided.

Reply:
we could.. but it's not how this method works... its more like a static checking , to avoid fibers, collections and more logic inside... if you are suggesting this improve there are a LOT OF MORE work to do where.... my intention on this PR is make more explicit the ifs... room.t !== 'd' doesn't make much sense... BUT !room.canChangeXValue(asda)